src/lib/utils/resolvedReplyType.ts

```diff
@@ -12,3 +20,24 @@ export default function getCaseResolvedReplyType(replyType: UpperCaseReplyType |
     upperCase: <UpperCaseReplyType>replyType.toUpperCase(),
   };
 }
+
+type UpperCaseThreadReplySelectType = typeof ThreadReplySelectType[keyof typeof ThreadReplySelectType];
+type LowerCaseThreadReplySelectType = Lowercase<UpperCaseThreadReplySelectType>;
+
+/**
+ * Dashboard UIKitConfig's threadReplySelectType is consisted of all lowercase letters,
+ * we need to convert it into uppercase ones(or vice versa)
+ * when we pass the value to internal components
+ *  - 'thread' <-> 'THREAD'
+ *  - 'parent' <-> 'PARENT'
+ */
+export function getCaseResolvedThreadReplySelectType(threadReplySelectType: UpperCaseThreadReplySelectType | LowerCaseThreadReplySelectType): {
```

Comment:
nit -> 
I would recommend having `toLowerCase`  & `toUpperCase` functions ->
these fns are a bit too specific, but they do very simple/generic work tho?

Reply:
What do you think?